### PR TITLE
Jacobi EVD without matrix mult

### PR DESCRIPTION
**Pull request by MartinKarp**
My implementation of the EVD, I manage to not do a triple loop for the matrix multiplication, but please double check that is is correct. The matrix A, which I think should be symmetric at all times, is not always symmetric, but it is decreasing in all spots except the diagonal and converges to the correct eigenvalues. Do not know why, but it is something we should discuss. I have one or two tests, but I would like to make them a little better before uploading them.

### CODE REVIEW

File: src/evd.cpp [hunk 1]
```diff
@@ -19,3 +19,90 @@ void sym_schur2(const double* const X, int n, int p, int q, double* c, double* s
...
+                double abs_ij = abs(A[i*n+j]);
+                if(abs_ij > abs_a){
+                    abs_a = abs_ij;
+                    p = i;
+                    q = j;
+                }
+            }
+        }
+
+        sym_schur2(A,n,p,q,&c,&s);
```
Comment (eozd) [inline src/evd.cpp]: Could you use the new ```jacobi_sym_coeffs``` function here? They have the same functionality, but the new one is more generic. We can delete this function at a later stage.

**Comment by eozd** [comment]
I believe we should change the title of the PR to one that describes what is implemented here so that it will be easier to reference it later if needed.

**Comment by ktrianta** [comment]
It would be nice if the commit messages are also a bit more descriptive. For example:

- Implement this as part of that
- Fix this in that
- Style and whitespace changes
- Add that functionality

**Comment by MartinKarp** [comment]
> It would be nice if the commit messages are also a bit more descriptive. For example:
> 
>     * Implement this as part of that
> 
>     * Fix this in that
> 
>     * Style and whitespace changes
> 
>     * Add that functionality

Sounds good!

Can I add this now afteerwards?

**Comment by ktrianta** [comment]
> > It would be nice if the commit messages are also a bit more descriptive. For example:
> > ```
> > * Implement this as part of that
> > 
> > * Fix this in that
> > 
> > * Style and whitespace changes
> > 
> > * Add that functionality
> > ```
> 
> Sounds good!
> 
> Can I add this now afteerwards?

It is possible if you do git rebase, rename and then push the changes.

File: src/evd.cpp (hunk 1 above)
Comment (MartinKarp) [reply src/evd.cpp]: will do!

**Comment by MartinKarp** [comment]
> > > It would be nice if the commit messages are also a bit more descriptive. For example:
> > > ```
> > > * Implement this as part of that
> > > 
> > > * Fix this in that
> > > 
> > > * Style and whitespace changes
> > > 
> > > * Add that functionality
> > > ```
> > 
> > 
> > Sounds good!
> > Can I add this now afteerwards?
> 
> It is possible if you do git rebase, rename and then push the changes.

Alright, I have done that now. However, I could not change the "I just want to pull" message. But I will make better commit message for future commits.

**Comment by eozd** [comment]
> > > > It would be nice if the commit messages are also a bit more descriptive. For example:
> > > > ```
> > > > * Implement this as part of that
> > > > 
> > > > * Fix this in that
> > > > 
> > > > * Style and whitespace changes
> > > > 
> > > > * Add that functionality
> > > > ```
> > > 
> > > 
> > > Sounds good!
> > > Can I add this now afteerwards?
> > 
> > 
> > It is possible if you do git rebase, rename and then push the changes.
> 
> Alright, I have done that now. However, I could not change the "I just want to pull" message. But I will make better commit message for future commits.

You can try squashing all of these commits to a single commit

**Pull request merged**